content/en/blog/_posts/2024-08-20-websocket-transition.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: blog
+title: 'Streaming Transitions from SPDY to WebSockets'
+date: 2024-08-20
+slug: websockets-transition
+author: >
+  [Sean Sullivan](https://github.com/seans3) (Google)
+  [Shannon Kularathna](https://github.com/shannonxtreme) (Google)
+---
+
+In Kubernetes 1.31, by default kubectl now uses the WebSocket protocol
+instead of SPDY for streaming.
+
+This post describes what these changes mean for you and why these streaming APIs
+matter.
+
+## Streaming APIs in Kubernetes
+
+In Kubernetes, specific endpoints that are exposed as an HTTP or RESTful
+interface are upgraded to streaming connections, which require a streaming
+protocol. Unlike HTTP, which is a request-response protocol, a streaming
+protocol provides a persistent connection that's bi-directional, low-latency,
+and lets you interact in real-time. Streaming protocols support reading and
+writing data between your client and the server, in both directions, over the
+same connection. This type of connection is useful, for example, when you create
+a shell in a running container from your local workstation and run commands in
+the container.
+
+## Why the streaming protocol is changing
+
+Currently, Kubernetes uses the SPDY/3.1 protocol for upgraded streaming
```

Comment:
This will be published _after_ the release has happened, so: (eg)
```suggestion
Before the v1.31 release, Kubernetes used the SPDY/3.1 protocol by default when upgrading
existing connections to streaming
```

Reply:
Fixed.